layers: Fix format_feature_2 format features

sfricke-samsung opened the pull request:
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/3736

- `VUID-*-OpTypeImage-06423` and `VUID-*-OpTypeImage-06424` should be checking image view format
    - spec fix - https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/5041
- `VUID-*-None-06479` also says "image view" but was checking the image format features
- Added a slight more information at error message time

cc @djdeath

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 9049.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 6148 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 6148 passed.

djdeath commented:
:+1: Not sure if we need to wait for the spec change.

ncesario-lunarg reviewed (approved):
LGTM! Could we add a test case?

ncesario-lunarg commented:
I'm good with merging this now. What do you think @sfricke-samsung?

sfricke-samsung commented:
> Not sure if we need to wait for the spec change.

Due to being a clear oversight I see no issue merging now

> Could we add a test case?

I don't have a device with 1.3 or `VK_KHR_format_feature_flags2` currently to write a test

llandwerlin-intel commented:
> > Could we add a test case?
> 
> I don't have a device with 1.3 or `VK_KHR_format_feature_flags2` currently to write a test

Lunarg has a mock device that's run on CI, that could be enough run the test : https://github.com/KhronosGroup/Vulkan-Tools/tree/master/icd

sfricke-samsung commented:
> Lunarg has a mock device that's run on CI, that could be enough run the test

I figured it would be... ya let me that, both nice to get my mock ICD setup working on my current machine as well as I like regression tests a lot (same with https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/3741 will add a test for that as well)

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 10132.

sfricke-samsung commented:
So added a tests... but required a bit more work

- Updated the Device Profile layer to support `VkFormatProperties3`
- Added doc how to use it :muscle: 
- Had to update `IsCompatible` check in testing framework to support `VkFormatFeatureFlags2`

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 6165 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 6165 failed.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 10236.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 6166 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 6166 failed.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 10622.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 6168 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 6168 failed.

Tony-LunarG commented on tests/vklayertests_pipeline_shader.cpp:
```diff
@@ -2224,7 +2224,7 @@ TEST_F(VkLayerTest, MissingStorageImageFormatWriteForFormat) {
             (fmt_props_3.optimalTilingFeatures & VK_FORMAT_FEATURE_2_STORAGE_IMAGE_BIT_KHR) != 0;
         const bool has_read_without_format =
-            (fmt_props_3.optimalTilingFeatures & VK_FORMAT_FEATURE_2_STORAGE_READ_WITHOUT_FORMAT_BIT_KHR) != 0;
+            (fmt_props_3.optimalTilingFeatures & VK_FORMAT_FEATURE_2_STORAGE_WRITE_WITHOUT_FORMAT_BIT_KHR) != 0;
```
Should this be READ_WITHOUT_FORMAT or is the variable misnamed?

sfricke-samsung replied:
so this was a copy and paste error with the MissingStorageImageFormat**Read**ForFormat test... but I realize I didn't update the variable name as well. Can do that too

Tony-LunarG commented on tests/positive/pipeline.cpp:
```diff
@@ -6391,6 +6391,168 @@ TEST_F(VkPositiveLayerTest, LineTopologyClasses) {
+                                                                 &format_props)) {
+        printf("%s Device will not be able to initialize image skipped.\n", kSkipPrefix);
+        return;
+    }
```
Of the devices we have in CI that have the extensions needed to run the test, none of them pass this GetPhysicalDeviceImageFormatProperties test.  @sfricke-samsung did you have hardware that made it by this check?

sfricke-samsung replied:
I don't have hardware, instead had to update the Devie Profile layer so I could `devsim` my way through this test

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build queued with queue ID 13951.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 6201 running.

ci-tester-lunarg commented:
CI Vulkan-ValidationLayers build # 6201 passed.

Tony-LunarG reviewed (approved):
LGTM - thanks for the rename.
